### PR TITLE
feat: Integrate ProcessProposal [BLO-840]

### PR DESCRIPTION
**Pull request by g3ndo9kar9**
## In This PR
- I integrate the process of updating dydx's ProcessProposal handler to be amenable to slinky
- Specifically
   - I move all `error` defs to `/process/errors` pkg
   - I implement an `UpdateMarketPriceTxDecoder` interface and use that to decode `UpdateMarketPricesTx`s in the ProcessProposal handler
        - I implement the original logic of dydx, and slinky specific logic as implementations of this interface
    - I also update the DecodeProcessProposalTxs` in `transactions.go` to account for the injected extended-commit tx (if vote-extensions are enabled)
## Changed Files
- `/app/process/prices` -> directory contains impls + intefaces
- `/app/process/transactions.go` -> I update the decoding logic of injected txs to account for the slinky injected txs
- All other changes are minor (changing error defs / types, mocks, testing)

### CODE REVIEW

File: protocol/app/process/errors/errors.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package errors
```
Comment (Eric-Warehime) [inline protocol/app/process/errors/errors.go]: Moving errors to a separate package is just creating unnecessary changes. All of the usage is still in the process package so I don't really see a need to move it.

File: protocol/app/process/prices/expected_keepers.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package prices
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	pricestypes "github.com/dydxprotocol/v4-chain/protocol/x/prices/types"
+)
+
+// PricesKeeper defines the expected Prices keeper used for `DefaultMarketPriceDecoder`.
+type PricesKeeper interface {
```
Comment (Eric-Warehime) [inline protocol/app/process/prices/expected_keepers.go]: This is basically already defined in process https://github.com/dydxprotocol/v4-chain/blob/83dd2c9df394351acd6340077cdb527661b5d1af/protocol/app/process/expected_keepers.go#L16

Adding more packages here seems to be creating a lot of churn/duplication.

**Review by Eric-Warehime — COMMENTED** [review]
Actual logic looks good. I think removing all the refactoring stuff will make the diff much smaller which to me is going to be really important when thinking about merging this back upstream.

**Review by Eric-Warehime — APPROVED** [review]

File: protocol/app/prepare/prices/slinky_price_update_generator.go [hunk 3]
```diff
@@ -89,5 +91,14 @@ func (pug *SlinkyPriceUpdateGenerator) GetValidMarketPriceUpdates(ctx sdk.Contex
...
 	}
 
+	// sort the market-price updates
+	sort.Slice(msg.MarketPriceUpdates, func(i, j int) bool {
+		return msg.MarketPriceUpdates[i].MarketId < msg.MarketPriceUpdates[j].MarketId
+	})
+
+	if err := msg.ValidateBasic(); err != nil {
+		return nil, err
+	}
```
Comment (g3ndo9kar9) [inline protocol/app/prepare/prices/slinky_price_update_generator.go:101]: Ensure that messages returned pass ValidateBasic

File: protocol/app/process/slinky_market_price_decoder.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+func checkEqualityOfMarketPriceUpdate(expectedMsg, actualMsg *pricestypes.MsgUpdateMarketPrices) error {
+	// assert that the market-price updates are valid
+	if err := expectedMsg.ValidateBasic(); err != nil {
+		return InvalidMarketPriceUpdateError(err)
+	}
+
+	if err := actualMsg.ValidateBasic(); err != nil {
+		return InvalidMarketPriceUpdateError(err)
+	}
+	// assert len is correct
```
Comment (g3ndo9kar9) [inline protocol/app/process/slinky_market_price_decoder.go:95]: Can assume that MarketIds in the `MsgUpdateMarketPrices` are ordered
